src/test/java/core/basesyntax/service/TransitionParserImplTest.java

```diff
@@ -0,0 +1,69 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransition;
+import core.basesyntax.service.impl.OperationValidatorImpl;
+import core.basesyntax.service.impl.TransitionParserImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransitionParserImplTest {
+    private static List<String> data;
+    private static TransitionParser transitionParser;
+
+    @BeforeClass
+    public static void setUp() {
+        OperationValidator validator = new OperationValidatorImpl();
+        transitionParser = new TransitionParserImpl(validator);
+        data = new ArrayList<>();
+    }
+
+    @Before
+    public void beforeAll() {
+        data.clear();
+        data.add("type,fruit,quantity");
+    }
+
+    @Test
+    public void parseTransition_getFruit_ok() {
+        data.add("b,apple,100");
+        List<FruitTransition> expected = new ArrayList<>();
+        expected.add(new FruitTransition("b", new Fruit("apple"), 100));
+        List<FruitTransition> actual = transitionParser.parseTransition(data);
+        assertEquals(expected.get(0).getFruit(), actual.get(0).getFruit());
+        assertEquals(expected.get(0).getCount(), actual.get(0).getCount());
+        assertEquals(expected.get(0).getOperation(), actual.get(0).getOperation());
```

Comment:
we should also check `actual.size()`
but it would be better if we had overridden `equals` in `FruitTransition` class, 
so we could just write `assertEquals(expected, actual)`